src/io/s3_filesys.cc

```diff
@@ -1028,7 +1028,7 @@ void S3FileSystem::ListObjects(const URI &path, std::vector<FileInfo> *out_list)
     std::string canonical_uri;
     std::string canonical_querystring;
 
-    AddDefaultCanonicalHeaders(&canonical_headers, curr_time, s3_session_token_, payload, false);
+    AddDefaultCanonicalHeaders(&canonical_headers, curr_time, s3_session_token_, payload, true);
```

Comment:
Isn't payload empty when reading? Is this required? What scenario did you run into which failed?